app.rb

```diff
@@ -0,0 +1,157 @@
+      return if age == ''
+    end
+    print 'Classroom: '
+    classroom = gets.chomp.to_s
+    while classroom == ''
+      print 'Please enter a classroom or press enter to exit: '
+      classroom = gets.chomp.to_s
+      return if classroom == ''
+    end
+    persons.push(Student.new(classroom, age, name))
+    puts 'New Student has created successfully!'
+  end
+
+  def create_teacher(name)
+    print 'Age: '
+    age = gets.chomp.to_i
+    until age.integer?
+      print 'Please enter a number or press enter to exit: '
+      return if age == ''
+    end
+    print 'Specification: '
+    specification = gets.chomp.to_s
+    while specification == ''
+      print 'Please enter a specification or press enter to exit: '
+      specification = gets.chomp.to_s
+      return if specification == ''
+    end
+    persons.push(Teacher.new(specification, age, name))
+    puts 'New Teacher has been created successfully!'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp.to_s
+    while title == ''
+      print 'Please enter a title or press enter to exit: '
+      title = gets.chomp.to_s
+      return if title == ''
+    end
+    print 'Auhtor: '
+    author = gets.chomp.to_s
+    while author == ''
+      print 'Please enter an author or press enter to exit: '
+      author = gets.chomp.to_s
+      return if author == ''
+    end
+    books.push(Book.new(title, author))
+    puts 'New book has been added successfully'
+  end
+
+  def create_rental
+    print 'Date (yyyy-mm-dd): '
+    date = gets.chomp
+    rgx = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/
+    until rgx.match?(date)
+      print 'Please enter a valid date or press enter to exit: '
+      date = gets.chomp
+      return if (date = '')
+    end
+    print 'Book: '
+    book = gets.chomp
+    until book.instance_of?(Book)
+      print 'Please enter a valid book or press enter to exit: '
+      book = gets.chomp
+      return if (book = '')
+    end
+    print 'Book: '
+    person = gets.chomp
+    until person.instance_of?(Person)
+      print 'Please enter a valid person or press enter to exit: '
+      person = gets.chomp
+      return if (person = '')
+    end
+    rentals.push(Rental.new(date, book, person))
+    puts 'New rental has been added successfully'
+  end
+
+  def list_persons
+    if persons.empty?
+      puts 'No person created'
+      return
+    end
+    puts "Amount of persons #{persons.length}"
+    persons.each_with_index do |p, index|
+      if p.instance_of?(::Student)
+        puts "#{index + 1} ID: #{p.id}, Name: #{p.name}, Age: #{p.age}, Classroom: #{p.classroom.label}}"
+      else
+        puts "#{index + 1} ID: #{p.id}, Name: #{p.name}, Age: #{p.age}, Specialization: #{p.specialization}"
+      end
+    end
+  end
+
+  def list_books
+    if books.empty?
+      puts 'No book created'
+      return
+    end
+    puts "Amount of books #{books.length}"
+    books.each_with_index do |b, index|
+      puts b[id]
```

Comment:
- [ ] [REQUIRED] After creating a book, I am unable to list the books created. I get the error bellow:
<img width="948" alt="Screenshot 2023-04-14 at 21 58 11" src="https://user-images.githubusercontent.com/58692590/232154637-352220b1-30e9-4bb1-8978-c4d79ecef301.png">

### My Observation
The error message you're seeing suggests that the list_books method is looking for a variable called `id` that hasn't been defined or initialized.

Let's define `id` to get this code to work 💪🏽

Reply:
Thanks. fixed that issue.